server/external.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import "golang.org/x/oauth2"
+
+type ExternalServiceClient struct {
+	token *oauth2.Token
+}
+
+func (s *ExternalServiceClient) GetMe() (string, error) {
+	return "some user id for token " + s.token.AccessToken, nil
+}
```

Comment:
Note that this struct represents something that would be defined in an external library specific to your integration, that supports usage of OAuth2 tokens.